python/cuml/cluster/dbscan.pyx

```diff
@@ -168,21 +169,18 @@ class DBSCAN(Base):
     """
 
     def __init__(self, eps=0.5, handle=None, min_samples=5, verbose=False,
-                 max_mbytes_per_batch=None):
-        super(DBSCAN, self).__init__(handle, verbose)
+                 max_mbytes_per_batch=None, output_type='cupy'):
```

Comment:
Proposal to start discussion at least:
 Maybe `output_type=None` is the default, which means we set to the default output type, and default output type is configurable as a global setting. Then you can either set it globally or do a context manager for these like:
 ```
with cuml.default_output_as('numpy'):
        do_my_legacy_code_that_expects_numpy()
```
 and of course callers could still override per model if they wanted.

Reply:
I like that idea, will definitely think about it and will give us more freedom in case we wanna change defaults and overall seems elegant. Excellent idea!

Reply:
Would it make sense to make the default output type match the input type? As a user, I would think this would be more consistent with my intuition than setting a global type by default. Could we add an option like “input” that matches the input type?

Reply:
Only mirroring the input type is more limited than having a global configuration. I.e. it’s not hard to imagine where a user might be feeding a model format A when they want format B as output because that is better for another library or for their purposes (maybe they have gpu arrays but then will be using a cpu library afterwards). That said, there is no reason why that behavior cannot be the default, with the global configuration to override it, which gives the best of both worlds.

Reply:
I totally only read the first question and for some reason thought you meant only mirror the input, your suggestion of an input option is what I verbosely described :P. So yeah, a mirroring option totally makes sense and candidate for the default behavior in my opinion!